woodwork/tests/accessor/test_statistics.py

```diff
@@ -1240,6 +1240,40 @@ def test_describe_accessor_method(describe_df):
         assert expected_vals.equals(stats_df["latlong_col"].dropna())
 
 
+@patch.object(sys.modules["woodwork.statistics_utils._get_describe_dict"], "percentile")
+def test_percentile_func_usage(mock_percentile, describe_df):
+    nullable_numeric_ltypes = [Double, IntegerNullable, AgeNullable, AgeFractional]
+    non_nullable_numeric_ltypes = [Integer, Age]
+    # Test numeric columns with nullable ltypes
+    numeric_data = describe_df[["numeric_col"]]
+    for ltype in nullable_numeric_ltypes:
+        numeric_data.ww.init(
+            logical_types={"numeric_col": ltype},
+            semantic_tags={"numeric_col": "custom_tag"},
+        )
+        numeric_data.ww.describe()
+        assert not mock_percentile.called
+
+    # Test numeric with non-nullable ltypes
+    numeric_data = describe_df[["numeric_col"]].dropna()
+    for ltype in nullable_numeric_ltypes:
+        numeric_data.ww.init(
+            logical_types={"numeric_col": ltype},
+            semantic_tags={"numeric_col": "custom_tag"},
+        )
+        numeric_data.ww.describe()
+        assert mock_percentile.called
```

Comment:
By doing `mock_percentile.called` in a for loop, you're only really confirming that the first logical type has `percentile` called (since `mock_percentile.called` will stay `True` for the rest of the test). My recommendation would be to parametrize over the different nullable logical types; then, each time the test is created it has a fresh `mock_percentile`. 

Additionally, the check below with the non nullable types would run into a similar issue bc once the mock has been called, we don't learn anything from confirming that it's still called. I think it may just be simplest to make a totally separate test for the non nullable ltypes and only check that it's not called prior to the describe called and is called afterwards. I'm not totally in love with that, though, and would be open to other solutions!